Proteomics/ProteolyticDigestion/DigestionMotif.cs

```diff
@@ -127,32 +127,33 @@ public bool Fits(string sequence, int location)
                 }
             }
 
+            bool prevents = false;
             // check for preventing cleavage
             if (fits && PreventingCleavage != null)
             {
-                bool match = true;
-                for (int n = 0; n < PreventingCleavage.Length && match; n++)
+                prevents = true;
+                for (int n = 0; n < PreventingCleavage.Length && prevents; n++)
                 {
                     if (location + m + n > sequence.Length || location - PreventingCleavage.Length + 1 + n < 0)
                     {
-                        match = false;
+                        prevents = false;
                     }
                     else
                     {
-                        currentResidue = CutIndex != 0 ? sequence[location + m + n] : sequence[location - PreventingCleavage.Length + 1 + n];                      
+                        currentResidue = CutIndex != 0 ? sequence[location + m + n] : sequence[location - PreventingCleavage.Length + 1 + n];
                         if (!PreventingCleavage[n].Equals(currentResidue))
                         {
-                            match = false;
+                            prevents = false;
                         }
                     }
                 }
 
-                fits = match ? false : true;
+                fits = prevents ? false : true;
             }
 
-            return fits;
+            return (fits, prevents);
```

Comment:
can you leave the return type as bool and make line 154:

`return fits && !prevents;`

Reply:
or just get rid of the `prevents` bool entirely and just set `fits = false` on line 139, 146, etc

Reply:
plz just return one bool - `return fits && !prevents;`